pkg/client/unversioned/request.go

```diff
@@ -722,10 +757,8 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 			// TCP connection
 			defer resp.Body.Close()
 
-			retries++
-			if seconds, wait := checkWait(resp); wait && retries < maxRetries {
-				glog.V(4).Infof("Got a Retry-After %s response for attempt %d to %v", seconds, retries, url)
-				time.Sleep(time.Duration(seconds) * time.Second)
+			if seconds, wait := checkWait(resp); wait {
+				glog.V(4).Infof("Got a Retry-After %s response for attempt %d to %v, but will backoff exponentially instead.", seconds, url)
```

Comment:
Hmmm... Given that retries is associated with retry-after, currently, I guess the semantics is slightly different, since after ten or so we are approaching the 15 minute mark for retries...  
But yes there should be a limit.  
The original code actually acknowledges that timeout is more useful than retrying...
Maybe do 5 minutes or so as the default timeout for retrying?